### ISSUE TITLE
OpenXRVk utlis has unused argument warnings as errors

### ISSUE THREAD

**Issue by amzn-changml** [o3de/o3de-extras#687]
**Describe the bug**
The OpenXRVk gem has a unused argument warning as error in the `PrintXrError` function

**Assets required**
Jenkins o3de-extras release build log: https://jenkins.build.o3de.org/blue/organizations/jenkins/o3de-extras/detail/development/132/pipeline/811#step-821-log-736

Jenkins o3de release build log: https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE/branches/development/runs/4979/nodes/508/log/?start=0

**Steps to reproduce**
1. Build o3de with the release target: `cmake --build . --target ALL_BUILD --config release  -- /m /nologo`

**Expected behavior**
Build completes successfully without error

**Actual behavior**
An C2220 error as warning is encountered for the `PrintXrError` function

**Found in Branch**
Development branch

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) and [o3de/o3de-extras](https://github.com/o3de/o3de-extras) repositories**

c7b8a76003225731ba9f079ae29c5b26f6c13829

**Desktop/Device (please complete the following information):**

- Device: AWS EC2
- OS: Windows Server
- Version: 2022 (eqv to Windows 11)
- CPU: AMD EPYC 7R13
- GPU: N/A
- Memory 32GB

**Additional context**

Log snippet below:

```
[2024-04-23T00:42:37.355Z] D:/workspace/o3de-extras/Gems/OpenXRVk/Code/Source/OpenXRVkUtils.cpp(47,62): error C2220: the following warning is treated as an error [D:\workspace\o3de\build\windows\External\OpenXRVk-753c2f2e\Code\OpenXRVk.Static.vcxproj]
[2024-04-23T00:42:37.355Z]       void PrintXrError(const char* windowName, const XrResult error, const char* fmt, ...)
[2024-04-23T00:42:37.355Z]                                                                ^
[2024-04-23T00:42:37.355Z] D:/workspace/o3de-extras/Gems/OpenXRVk/Code/Source/OpenXRVkUtils.cpp(47,62): warning C4100: 'error': unreferenced formal parameter [D:\workspace\o3de\build\windows\External\OpenXRVk-753c2f2e\Code\OpenXRVk.Static.vcxproj]
[2024-04-23T00:42:37.355Z] D:/workspace/o3de-extras/Gems/OpenXRVk/Code/Source/OpenXRVkUtils.cpp(47,35): warning C4100: 'windowName': unreferenced formal parameter [D:\workspace\o3de\build\windows\External\OpenXRVk-753c2f2e\Code\OpenXRVk.Static.vcxproj]
[2024-04-23T00:42:37.355Z]       void PrintXrError(const char* windowName, const XrResult error, const char* fmt, ...)
[2024-04-23T00:42:37.355Z]                                     ^
```

**Comment by jhanca-robotecai** [comment]
Is this issue fixed in #688?

**Comment by amzn-changml** [comment]
Yes, this should be fixed. I'm closing this ticket as a result.
